src/messages/store/publish.ts

```diff
@@ -84,11 +132,82 @@ export async function Publish({
         APIServer,
     });
 
-    await SignAndBroadcast({
-        message: message,
-        account,
-        APIServer,
-    });
+    if (ItemType.ipfs == message.item_type) {
+        await SignAndBroadcast({
+            message: message,
+            account,
+            APIServer,
+        });
+    } else if (!fileObject) {
+        throw new Error("You need to specify a File to upload or a Hash to pin.");
+    } else {
+        return await sendMessage(
+            {
+                message: message,
+                account,
+                APIServer,
+                sync,
+            },
+            fileObject,
+        );
+    }
 
     return message;
 }
+
+async function processFileObject(fileObject: Blob | Buffer | null): Promise<string> {
+    if (!fileObject) throw new Error("fileObject is null");
+
+    if (fileObject instanceof Blob) {
+        fileObject = await blobToBuffer(fileObject);
+    }
+    return calculateSHA256Hash(fileObject);
+}
+
+type SignAndBroadcastConfiguration = {
+    message: BaseMessage;
+    account: Account;
+    APIServer: string;
+    sync: boolean;
+};
+
+async function sendMessage(configuration: SignAndBroadcastConfiguration, fileObject: Blob | Buffer) {
+    const form = new FormData();
+    const metadata = {
+        message: {
+            ...configuration.message,
+            signature: await configuration.account.Sign(configuration.message),
+        },
+        sync: configuration.sync,
+    };
+
+    form.append("file", fileObject);
+    form.append("metadata", JSON.stringify(metadata));
+
+    const response = await axios.post(`${stripTrailingSlash(configuration.APIServer)}/api/v0/storage/add_file`, form, {
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "multipart/form-data",
+        },
+    });
+    return response.data;
+}
+
+async function blobToBuffer(blob: Blob): Promise<Buffer> {
+    return new Promise<Buffer>((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = () => {
+            if (reader.result instanceof ArrayBuffer) {
+                resolve(Buffer.from(reader.result));
+            } else {
+                reject("Failed to convert Blob to Buffer.");
+            }
+        };
+        reader.readAsArrayBuffer(blob);
+    });
+}
+
+function calculateSHA256Hash(data: ArrayBuffer | Buffer): string {
+    const buffer = Buffer.from(data);
+    return new shajs.sha256().update(buffer).digest("hex");
+}
```

Comment:
Can you move these two functions into a new file in the [utils](https://github.com/aleph-im/aleph-sdk-ts/tree/main/src/utils) directory?